src/map.rs

```diff
@@ -477,6 +477,37 @@ where
         }
     }
 
+    /// Return the values for `N` keys. If any key is missing a value, or there
+    /// are duplicate keys, `None` is returned.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let mut map = indexmap::IndexMap::from([(1, 'a'), (3, 'b'), (2, 'c')]);
+    /// assert_eq!(map.get_many_mut([&2, &1]), Some([&mut 'c', &mut 'a']));
+    /// ```
+    pub fn get_many_mut<'a, 'b, Q: ?Sized, const N: usize>(
+        &'a mut self,
+        keys: [&'b Q; N],
+    ) -> Option<[&'a mut V; N]>
+    where
+        Q: Hash + Equivalent<K>,
+    {
+        let len = self.len();
+        let indices = keys.map(|key| self.get_index_of(key));
+
+        // Handle out-of-bounds indices with panic as this is an internal error in get_index_of.
+        for idx in indices {
+            let idx = idx?;
+            if idx >= len {
+                panic!("Index is out of range! Got '{idx}' but length is '{len}'")
```

Comment:
Well, since this is internal "shouldn't happen" territory, I would also be fine with letting such bugs fall through to `get_many_index_mut -> None`. Maybe just add a `debug_assert!(idx < len)` for good measure.

Reply:
Okay, sure, that makes sense!